app/models/shortener/shortened_url.rb

```diff
@@ -4,7 +4,7 @@ class Shortener::ShortenedUrl < ActiveRecord::Base
 
   validates :url, presence: true
 
-  around_create :generate_unique_key
+  before_create :generate_unique_key
```

Comment:
this callback is used for cases like `Shortener::ShortenedUrl.create` but your fix does not cover it which is highlighted with tests changes

Reply:
I've added one test example, WDYT?

Reply:
this fix does not work with `Shortener::ShortenedUrl.create` (which I think is expected) and that's why you have changed [this](https://github.com/jpmcgrath/shortener/pull/109/files#diff-89fe41dcc7d8198998f4d0f2e1bcaab8L129 )